### ISSUE TITLE
Remove ALT interface in 4.0

### ISSUE THREAD

**Issue by daverodgman** [Mbed-TLS/mbedtls#8149]
Remove all references to the ALT interface, i.e. things guarded by `#if defined(MBEDTLS_xxx_ALT)` in `include/mbedtls`, and clean up library and tests accordingly.

Parts of the cleanup could be deferred to later in the 4.x lifecycle if parts of this are tricky for some reason.

Exceptions:
- the platform interface is out of scope for this issue. Changes to the platform interface are covered by #8108
- threading: we will revisit this when the current threading work completes. @yanesca @paul-elliott-arm do you have any comments on this?
- timing - maybe we will need to retain this, or it may be in scope of #8108

**Comment by yanesca** [comment]
I agree that threading should be handled as part of https://github.com/Mbed-TLS/mbedtls/issues/8108 as I am not aware of any `_ALT` related to that.

I would imagine that in #8108 we would simplify the configuration of the platform interfaces and the impact on how they are used in the library would be limited. I think we should be able to do both in parallel with limited effort lost on conflicts.

**Comment by tom-cosgrove-arm** [comment]
Note: this should be for cryptographic `_ALT`s (for which there is the alternative of providing a PSA driver) not for things such as `MBEDTLS_PLATFORM_ZEROIZE_ALT`, for which there is currently no alternative

**Comment by gilles-peskine-arm** [comment]
Platform ALTs are covered by https://github.com/Mbed-TLS/mbedtls/issues/8108.

**Comment by gilles-peskine-arm** [comment]
As of today at ab0af45d11db04bd6e42ccfabe1b8d8deaae3243, there are no more crypto module ALT and no more ECP/AES ALT. So this task is done.
